src/state/swaps/swapsStore.ts

```diff
@@ -162,20 +162,42 @@ export const swapsStore = createRainbowStore<SwapsState>(
     getRecentSwapsByChain: (chainId: ChainId) => {
       const { recentSwaps } = get();
 
-      const chainSwaps = recentSwaps.get(chainId) || [];
-      return chainSwaps.sort((a, b) => b.swappedAt - a.swappedAt);
+      return recentSwaps.get(chainId) || [];
     },
     addRecentSwap(asset) {
       const { recentSwaps, latestSwapAt } = get();
       const now = Date.now();
       const chainId = asset.chainId;
       const chainSwaps = recentSwaps.get(chainId) || [];
 
-      const updatedSwaps = [...chainSwaps, { ...asset, swappedAt: now }].slice(-3);
+      const [latestSwap] = chainSwaps;
+
+      // Check if the most recent swap is the same as the incoming asset
+      if (latestSwap && latestSwap.uniqueId === asset.uniqueId) {
+        latestSwapAt.set(chainId, now);
+        recentSwaps.set(chainId, [
+          ...chainSwaps.slice(1),
+          {
```

Comment:
is `recentSwaps` supposed to be ordered from "newest" to "oldest" ?
confirming why we are setting the latest swap after `...chainSwaps.slice(1)` in the array

Reply:
We're only ordering when displaying the section now. `filterList` has built in methods to sort. The reason I moved it there was because the filterList function wasn't working with presorted items.

Reply:
So technically we can put the array it any order here, as it doesn't matter.

Reply:
how can we rely on `const [latestSwap] = chainSwaps;` being the actual latest swap ?

Reply:
oh true good eye. Let me grab it by the `swappedAt`